### ISSUE TITLE
`build/Generated/Types.hs` "Not in scope" error when creating a table without an `id` primary key

### ISSUE THREAD

**Issue by stianlagstad** [digitallyinduced/ihp#696]
Ref [this thread](https://ihpframework.slack.com/archives/C01DQE0F4F8/p1610359751035000) on the IHP Slack.

Steps to reproduce:
1. `ihp-new ihp-table-without-primary-key-error`
2. `cd ihp-table-without-primary-key-error`
3. `./start`
4. Open <http://localhost:8001/> and create the table `first`, and keep the `id` column that's added automatically.
5. Create the table `second`, and keep the `id` column for now
6. Create a column `second.first_id` and have it reference `first.id`
7. Remove the `second.id` column and see the error appear in the console:
```
build/Generated/Types.hs:43:87: error:
    Not in scope: type constructor or class ‘Second’
    Perhaps you meant ‘Record’ (imported from IHP.ModelSupport)
   |
43 |         pure $ First id (QueryBuilder.filterWhere (#firstId, id) (QueryBuilder.query @Second)) def
   |                                                                                       ^^^^^^

build/Generated/Types.hs:56:58: error:
    Not in scope: type constructor or class ‘Second’
    Perhaps you meant one of these:
      type variable ‘second’ (line 56),
      ‘Record’ (imported from IHP.ModelSupport)
   |
56 | type instance Include "second" (First' second) = First' [Second]
   |                                                          ^^^^^^
Failed, three modules loaded.
```

Repository with the error reproduced: <https://github.com/stianlagstad/ihp-table-without-primary-key-error/>

**Comment by fegu** [comment]
There are a few known bugs when one removes stuff in the schema designer. For instance removing a foreign key column will just remove the table column, not remove the foreign key constraint definition. The schema code view comes in handy.

**Comment by stianlagstad** [comment]
Thanks for responding @fegu! [The schema](https://github.com/stianlagstad/ihp-table-without-primary-key-error/blob/f13acb339806e1a135d852e15b29cc83d2014928/Application/Schema.sql) looks as I expect after step 7:
```
-- Your database schema. Use the Schema Designer at http://localhost:8001/ to add some tables.
CREATE TABLE "first" (
    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY NOT NULL
);
CREATE TABLE "second" (
    first_id UUID NOT NULL
);
ALTER TABLE "second" ADD CONSTRAINT second_ref_first_id FOREIGN KEY (first_id) REFERENCES "first" (id) ON DELETE NO ACTION;
```

So I don't think this has to do with a schema designer issue. Or am I misinterpreting you?

**Comment by fegu** [comment]
Did you remember to click primary key on the column in step 6? IHP does not like tables with no primary key.

**Comment by stianlagstad** [comment]
No, as I did not want that to be a primary key. As I read [this message](https://ihpframework.slack.com/archives/C01DQE0F4F8/p1610753657003900?thread_ts=1610359751.035000&cid=C01DQE0F4F8) from Daniel (and the following between Daniel and Marc), my intention for this issue is to a) have IHP allow tables without primary keys, and/or b) have the code that should've triggered [this error](https://ihpframework.slack.com/archives/C01DQE0F4F8/p1610753767004100?thread_ts=1610359751.035000&cid=C01DQE0F4F8) fixed.
